snap/validate.go

```diff
@@ -149,6 +237,14 @@ func validateField(name, cont string, whitelist *regexp.Regexp) error {
 	return nil
 }
 
+func validateAppSocket(socket *SocketInfo) error {
```

Comment:
can you make this one be `func (*SocketInfo) Validate() error`?

Reply:
I can, but the whole point of this module is to validate structs. The same reasoning could apply to `Info.Validate()`

Reply:
yes, and i'll be doing that refactor some day. I guess I can do this one at that point as well.